src/Compilers/Core/Portable/CodeGen/PrivateImplementationDetails.cs

```diff
@@ -110,8 +113,22 @@ internal void Freeze()
             }
 
             // Sort data fields
-            _orderedMappedFields = _mappedFields.Values.OrderBy((x, y) => x.Name.CompareTo(y.Name)).AsImmutable();
-            _orderedSynthesizedMethods = _synthesizedMethods.OrderBy(kvp => kvp.Key).Select(kvp => kvp.Value).AsImmutable();
+            ArrayBuilder<SynthesizedStaticField> fieldsBuilder = ArrayBuilder<SynthesizedStaticField>.GetInstance(_mappedFields.Count);
+            fieldsBuilder.AddRange(_mappedFields.Values.OrderBy((x, y) => x.Name.CompareTo(y.Name)));
+            if (_mvidField != null)
+            {
+                fieldsBuilder.Add(_mvidField);
```

Comment:
If that's not the intended behavior, consider building up the `ArrayBuilder` without order and then using `Sort`.